pandas/core/index.py

```diff
@@ -2580,7 +2580,10 @@ def drop(self, labels, errors='raise'):
     @Appender(_shared_docs['drop_duplicates'] % _index_doc_kwargs)
     def drop_duplicates(self, take_last=False):
         result = super(Index, self).drop_duplicates(take_last=take_last)
-        return self._constructor(result)
+        if self.name is not None:
```

Comment:
Yes. OK, I can do  that.

Alternatively, it would be great if the constructors of all ``Index`` subclasses (other than ``MultiIndex``) accepted a ``names`` list of length one and used it to set ``name``. That way one could always just set ``names`` without worrying about it.